code/datums/api.dm

```diff
@@ -0,0 +1,358 @@
+//
+// This file contains the API functions for the serverside API
+//
+
+//Init the API at startup
+/hook/startup/proc/setup_api()
+  for (var/path in typesof(/datum/topic_command) - /datum/topic_command)
+    var/datum/topic_command/A = new path()
+    topic_commands[A.name] = A
+
+  return 1
+
+//API Boilerplate
+/datum/topic_command
+  var/name = ""
+  var/statuscode = null
+  var/response = ""
+/datum/topic_command/proc/run_command(queryparams)
+  return 1
+
+//Ping Test
+/datum/topic_command/ping
+  name = "ping"
+/datum/topic_command/ping/run_command(queryparams)
+  var/x = 1
+  for (var/client/C)
+    x++
+  statuscode = "200"
+  response = x
+  return 1
+
+//Player Count
+/datum/topic_command/count_players
+  name = "count_players"
+/datum/topic_command/count_players/run_command(queryparams)
+  var/n = 0
+  for(var/mob/M in player_list)
+    if(M.client)
+      n++
+
+  statuscode = "200"
+  response = n
+  return 1
+
+//Admin Count
+/datum/topic_command/count_admin
+  name = "count_admin"
+/datum/topic_command/count_admin/run_command(queryparams)
+  var/n = 0
+  for (var/client/client in clients)
+    if (client.holder && client.holder.rights & (R_ADMIN))
+      n++
+
+  statuscode = "200"
+  response = n
+  return 1
+
+//Mod Count
+/datum/topic_command/count_mod
+  name = "count_mod"
+/datum/topic_command/count_mod/run_command(queryparams)
+  var/n = 0
+  for (var/client/client in clients)
+    if (client.holder && (client.holder.rights & R_MOD) && !(client.holder.rights & R_ADMIN))
+      n++
+
+  statuscode = "200"
+  response = n
+  return 1
+
+//CCIA Count
+/datum/topic_command/count_ccia
+  name = "count_ccia"
+/datum/topic_command/count_ccia/run_command(queryparams)
+  var/n = 0
+  for (var/client/client in clients)
+    if (client.holder && (client.holder.rights & R_CCIAA) && !(client.holder.rights & R_ADMIN))
+      n++
+
+  statuscode = "200"
+  response = n
+  return 1
+
+//Player Ckeys
+/datum/topic_command/player_list_ckey
+  name = "player_list_ckey"
+/datum/topic_command/player_list_ckey/run_command(queryparams)
+  var/list/players = list()
+  for (var/client/C in clients)
+    players += C.key
+
+  statuscode = "200"
+  response = list2params(players)
+  return 1
+
+//Char Names //Doesnt work properly --> Look into playerpanel how its done
+/datum/topic_command/char_list
+  name = "char_list"
+/datum/topic_command/char_list/run_command(queryparams)
+  if (!ticker)
+    statuscode = "500"
+    response = "Game not started yet!"
+    return 1
+
+  var/list/mobs = list()
+  for(var/mob/M in mobs)
```

Comment:
This might not work due to ambiguous naming. You have a local variable called mobs, and you have a global list you're presumably trying to access that is also called mobs. It may be trying to read the local one instead.